### PR TITLE
Decomission Experimental Terminal/Adapt Tasks for use with gRPC Runner

### PR DESCRIPTION
**Pull request by mxsdev**
Solves #290

Currently I didn't try to support tasks in non-grpc runner environments. ~~Tasks also don't use the same environment as the kernel (they don't have any environment persistence currently).~~

A lot of the stuff for tasks is copied from the runner executor. I'd like to unify the two but it's not super trivial. I see this is as a good POC for now, though I'm happy to take the time to solidify it if desired.

### CODE REVIEW

File: src/extension/provider/runmeTask.ts [hunk 1]
```diff
@@ -63,36 +86,51 @@ export class RunmeTaskProvider implements TaskProvider {
...
-    const definition: RunmeTaskDefinition = {
-      type: 'runme',
-      filePath,
-      command,
-      closeTerminalOnSuccess,
-      isBackground,
-      cwd
+    const isBackground = options.isBackground || background
+
+    const runOpts: RunProgramOptions = {
```
Comment (sourishkrout) [inline src/extension/provider/runmeTask.ts:106]: Do we have to "seed" the `envs` here with the host processes? Or does the runner take of that?

**Review by sourishkrout — CHANGES_REQUESTED** [review]
Possible clarification.

File: src/extension/provider/runmeTask.ts (hunk 1 above)
Comment (mxsdev) [reply src/extension/provider/runmeTask.ts:106]: Oh yeah, we have to seed it, good call.

In the future I'd like to integrate this with the existing session.

Comment (sourishkrout) [reply src/extension/provider/runmeTask.ts:106]: Makes sense.

**Comment by mxsdev** [comment]
Added support for environments @sourishkrout

File: src/extension/provider/runmeTask.ts (hunk 1 above)
Comment (mxsdev) [reply src/extension/provider/runmeTask.ts:109]: Added support for environments and env population in 452eb96 @sourishkrout

**Review by sourishkrout — APPROVED** [review]
> A lot of the stuff for tasks is copied from the runner executor. I'd like to unify the two but it's not super trivial. I see this is as a good POC for now, though I'm happy to take the time to solidify it if desired.

👍 Good call. Let's circle back when the time's right.

**Pull request merged**